common/src/main/java/com/microsoft/identity/common/internal/net/ObjectMapper.java

```diff
@@ -47,6 +58,135 @@
+                                in.beginObject();
+                            }
+
+                            public void endObject() throws IOException {
+                                in.endObject();
+                            }
+
+                            public boolean hasNext() throws IOException {
+                                return in.hasNext();
+                            }
+
+                            public JsonToken peek() throws IOException {
+                                return in.peek();
+                            }
+
+                            public String nextName() throws IOException {
+                                final String name = in.nextName();
+                                lastName = name;
+                                return name;
+                            }
+
+                            public String nextString() throws IOException {
+                                return in.nextString();
+                            }
+
+                            public boolean nextBoolean() throws IOException {
+                                return in.nextBoolean();
+                            }
+
+                            public void nextNull() throws IOException {
+                                in.nextNull();
+                            }
+
+                            public double nextDouble() throws IOException {
+                                return in.nextDouble();
+                            }
+
+                            public long nextLong() throws IOException {
+                                return in.nextLong();
+                            }
+
+                            public int nextInt() throws IOException {
+                                return in.nextInt();
+                            }
+
+                            public void close() throws IOException {
+                                in.close();
+                            }
+
+                            public void skipValue() throws IOException {
+                                JsonToken token = in.peek();
+                                if (token == JsonToken.STRING) {
+                                    otherKeys.put(lastName, in.nextString());
+                                } else {
+                                    in.skipValue();
+                                }
+                            }
+
+                            @NonNull
+                            @Override
+                            public String toString() {
+                                return in.toString();
+                            }
+
+                            public String getPath() {
+                                return in.getPath();
+                            }
+                        };
+                        T output = adapter.read(jsonReader);
+                        ((IHasExtraParameters) output).setExtraParameters(Collections.unmodifiableMap(otherKeys).entrySet());
```

Comment:
Would it make sense to use a bounded wildcard on the input type param to make sure this cast is going to work? Not sure my syntax is perfect here but something like

```java
TypeToken<T extends IHasExtraParameters> type
```

Reply:
We should only be in this block:

if (IHasExtraParameters.class.isAssignableFrom(type.getRawType())) {

So unless json breaks the contract...